phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexRebuildTaskIT.java

```diff
@@ -0,0 +1,200 @@
+                            + " CONSTRAINT NAME_PK PRIMARY KEY (%s PK2)" + " ) %s";
+            conn.createStatement().execute(generateDDL(ddlFormat));
+            conn.commit();
+            // Create a view
+            String viewName = generateUniqueName();
+            String viewDDL = "CREATE VIEW " + viewName + " AS SELECT * FROM " + baseTable;
+            viewConn.createStatement().execute(viewDDL);
+
+            // Create index
+            String indexName = generateUniqueName();
+            String idxSDDL = String.format("CREATE INDEX %s ON %s (V1)", indexName, viewName);
+
+            viewConn.createStatement().execute(idxSDDL);
+
+            // Insert rows
+            viewConn.createStatement()
+                    .execute(String.format("UPSERT INTO %s VALUES('%s', '%s', '%s')", viewName, "x", "y", "z"));
+            viewConn.commit();
+
+            String data = "{GusWorkId:abc, DisableBefore:true}";
+
+            // Add a task to System.Task to build indexes
+            TaskRegionObserver.addTask(conn.unwrap(PhoenixConnection.class), PTable.TaskType.INDEX_REBUILD,
+                    isMultiTenant? TENANT1 : null, null, viewName, indexName,
+                    PTable.TaskStatus.CREATED.toString(), data, null, null, true);
+
+            // Run IndexRebuildTask
+            TaskRegionObserver.IndexRebuildTask task =
+                    new TaskRegionObserver.IndexRebuildTask(
+                            TaskRegionEnvironment, QueryServicesOptions.DEFAULT_TASK_HANDLING_MAX_INTERVAL_MS);
+
+            task.run();
+
+            String viewIndexTableName = "_IDX_" + baseTable;
+            ConnectionQueryServices queryServices = conn.unwrap(PhoenixConnection.class).getQueryServices();
+            int count = getUtility().countRows(queryServices.getTable(Bytes.toBytes(viewIndexTableName)));
+            assertTrue(count == 1);
+
+
+            // Remove index contents and try again
+            Admin admin = queryServices.getAdmin();
+            TableName tableName = TableName.valueOf(viewIndexTableName);
+            admin.disableTable(tableName);
+            admin.truncateTable(tableName, false);
+
+            data = "{GusWorkId:abc}";
+
+            // Add a new task (update status to created) to System.Task to rebuild indexes
+            TaskRegionObserver.addTask(conn.unwrap(PhoenixConnection.class), PTable.TaskType.INDEX_REBUILD,
+                    isMultiTenant? TENANT1 : null, null, viewName, indexName,
+                    PTable.TaskStatus.CREATED.toString(), data, null, null, true);
+            task.run();
+
+            Table systemHTable= queryServices.getTable(Bytes.toBytes("SYSTEM."+PhoenixDatabaseMetaData.SYSTEM_TASK_TABLE));
+            count = getUtility().countRows(systemHTable);
+            assertTrue( count == numOfRuns);
+
+            // Check task status and other column values.
+            ResultSet rs = conn.createStatement().executeQuery("SELECT * " +
```

Comment:
Seems like this would be useful to have as a utility function somewhere. In general it's good to break up long functions into helper functions with descriptive names when there's a discrete unit of work.